js/src/elements/form-behavior.ts

```diff
@@ -268,8 +269,10 @@ export class FormBehaviorElement extends HTMLElement {
 				errorAlert.focus();
 			}
 		} finally {
-			this.submitting = false;
-			setBusySubmitButton(form, this.submitting);
+			if (!this.getAttribute('navigation') || !isLoading) {
+				this.submitting = false;
+				setBusySubmitButton(form, this.submitting);
+			}
 		}
```

Comment:
with isNavigating I think the original code could be updated to:

```diff
  } finally {
    this.submitting = false;
-   setBusySubmitButton(form, this.submitting);
+   setBusySubmitButton(form, this.submitting || isNavigating);
  }
```

Reply:
Thanks for the feedback!

I tested these changes out and everything seems to be working ok except for buttons that submit but don't reload the page. Example: on Q&A v2, the answer rating buttons don't reload the page, but they do prompt UI changes. With these updates, the button keeps the is-loading icon since this use case would return isNavigating = false.

Reply:
If I update line 271 to `this.submitting = this.submitting === isNavigating;`, all the buttons I was testing seem to be working

Reply:
apologies- this should have been:
```diff
  } finally {
-   this.submitting = false;
-   setBusySubmitButton(form, this.submitting);
+   this.submitting = !isNavigating;
+   setBusySubmitButton(form, this.submitting);  // I mistakenly put `|| isNavigating` here in the previous comment
  }
```

Reply:
Should it be `this.submitting = isNavigating`? 

With `this.submitting = !isNavigating`, on answer rating where isNavigating is false, it's not removing the is-loading class.

With `this.submitting = isNavigating`, everything seems to be working except when posting a new question, which uses "follow"

Reply:
I have pushed some updates based on your feedback and some testing done in qna v2 :)